No logs on exception in QuarkusTestResourceLifecycleManager.start

**Describe the bug**
Imagine having a QuarkusTestResourceLifecycleManager, which might throw an exception, if you have got a bug in your manager or haven't properly prepared your environment. If this manager throws an exception in the start method, no logs except the thrown exception are outputted on console.

**Expected behavior**
All logs, including already logged messages and the stack trace of the uncaught exception, are logged.

**Actual behavior**
Only the stack trace of the uncaught exception are logged.

**To Reproduce**
Steps to reproduce the behavior:
1. Create a test using a QuarkusTestResourceManager.
2. In the QuarkusTestResourceLifecycleManager, log something and throw a RuntimeException.
3. See the log not containing the logged entry from the manager.

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: 
 - Output of `java -version`: Linux xxx 4.4.0-17763-Microsoft #1217-Microsoft Mon May 05 16:09:00 PST 2020 x86_64 x86_64 x86_64 GNU/Linux
openjdk version "11.0.7" 2020-04-14
OpenJDK Runtime Environment (build 11.0.7+10-post-Ubuntu-3ubuntu1)
OpenJDK 64-Bit Server VM (build 11.0.7+10-post-Ubuntu-3ubuntu1, mixed mode)
 - GraalVM version (if different from Java): -
 - Quarkus version or git rev: 1.6.0.Final
 - Build tool (ie. output of `mvnw --version` or `gradlew --version`): Apache Maven 3.6.3

**Additional context**
https://quarkusio.zulipchat.com/#narrow/stream/187030-users/topic/No.20logs.20on.20exception.20in.20QTestResourceLifecycleManager.2Estart

I can't reproduce this. 

Here is what I got:

```
[INFO] Running org.acme.getting.started.GreetingResourceTest
I just got closed
[ERROR] Tests run: 2, Failures: 0, Errors: 1, Skipped: 1, Time elapsed: 2.703 s <<< FAILURE! - in org.acme.getting.started.GreetingResourceTest
[ERROR] testHelloEndpoint  Time elapsed: 0.045 s  <<< ERROR!
java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
Caused by: java.lang.reflect.InvocationTargetException
Caused by: java.lang.RuntimeException: Unable to start Quarkus test resource org.acme.getting.started.DummyTestResource@7fb53256
Caused by: java.lang.RuntimeException: Die!
```

--- a/test-framework/common/src/main/java/io/quarkus/test/common/QuarkusTestResourceLifecycleManager.java
+++ b/test-framework/common/src/main/java/io/quarkus/test/common/QuarkusTestResourceLifecycleManager.java
@@ -16,6 +16,9 @@ import java.util.Map;
  * Note that when using these with QuarkusUnitTest (rather than @QuarkusTest) they run
  * before the ClassLoader has been setup. This means injection may not work
  * as expected.
+ * <p>
+ * Due to the very early execution in the test setup lifecycle, logging does <b>not</b>
+ * work in such a manager.
  */
 public interface QuarkusTestResourceLifecycleManager {
 
@@ -52,7 +55,7 @@ public interface QuarkusTestResourceLifecycleManager {
     /**
      * If multiple Test Resources are specified,
      * this control the order of which they will be executed.
-     * 
+     *
      * @return The order to be executed. The larger the number, the later the Resource is invoked.
      */
     default int order() {

